lms/djangoapps/xblock_user_state/interface.py

```diff
@@ -0,0 +1,253 @@
+"""
+A baseclass for a generic client for accessing XBlock Scope.user_state field data.
+"""
+
+from abc import abstractmethod
+
+from contracts import contract, new_contract, ContractsMeta
+from opaque_keys.edx.keys import UsageKey
+from xblock.fields import Scope, ScopeBase
+
+new_contract('UsageKey', UsageKey)
+
+
+class XBlockUserStateClient(object):
+    """
+    First stab at an interface for accessing XBlock User State. This will have
+    use StudentModule as a backing store in the default case.
+
+    Scope/Goals:
+    1. Mediate access to all student-specific state stored by XBlocks.
+        a. This includes "preferences" and "user_info" (i.e. UserScope.ONE)
+        b. This includes XBlock Asides.
+        c. This may later include user_state_summary (i.e. UserScope.ALL).
+           - Though I'm fuzzy on where we're going in general with this and how
+             it relates to content.
+        d. This may include group state in the future.
+        e. This may include other key types + UserScope.ONE (e.g. Definition)
+           - I think this implies a per-user partition scheme and not a
+             user+course partition scheme.
```

Comment:
The bullet under `e.` is talking about Cassandra-specific musings. This looks like it was copy-pasted from my early wiki musings? If so, maybe check through it again to see if it still applies?